openssh/libsaltcli.jinja

```diff
@@ -10,6 +10,8 @@
 {%-   set cli = 'minion' %}
 {%- elif opts_cli == 'salt-call' %}
 {%-   set cli = 'ssh' if opts_masteropts_cli in ('salt-ssh', 'salt-master') else 'local' %}
+{%- elif opts_cli %}
+{%-   set cli = 'api' %}
```

Comment:
Just a slight concern that this is a little too loose (might end up catching something else we haven't come across yet and identifying it as `api`).

@danny-smit Would you mind providing a few examples of what shows up for `opts_cli` when using the Salt Client API?  I'm hoping that there is something common across all of the values, that we could use for matching instead.

Reply:
Here are all the reference I found about `__cli` in salt code:

```
~/src/saltstack/salt $ rgrep __cli salt/
salt/client/ssh/wrapper/state.py:        if __opts__["__cli"] == "salt-call"
salt/config/__init__.py:    opts["__cli"] = salt.utils.stringutils.to_unicode(os.path.basename(sys.argv[0]))
salt/config/__init__.py:    opts["__cli"] = salt.utils.stringutils.to_unicode(os.path.basename(sys.argv[0]))
salt/modules/git.py:        _salt_cli = __opts__.get("__cli", "")
salt/modules/state.py:        if __opts__.get("__cli", None) == "salt-call"
salt/modules/event.py:    if preload or __opts__.get("__cli") == "salt-call":
```

Reply:
@baby-gnu It looks like `__cli` really can be _anything_.  In that case, I think we're good with these changes.  Shall we wait for @danny-smit before merging or just go ahead?

Reply:
@myii `__cli` is, by default, the basename of `sys.argv[0]` and only overridden for `salt-ssh`, which is the only one which cause a problem with `config.get` options.

I think the `unknown` case could only happen when running from python command line, so I think it can be merged.